vespa/methods/detection/rcnn/model.py

```diff
@@ -18,23 +18,37 @@ def __init__(
         optimizer_name: str = 'adam',
         lr: float = 0.0001,
         weight_decay: float = 0.0001,
+        *args,
+        **kwargs,
     ):
-        super().__init__()
-        self.model = fasterrcnn_resnet50_fpn_v2(weights=weights)
-        in_features = self.model.roi_heads.box_predictor.cls_score.in_features
-        self.model.roi_heads.box_predictor = FastRCNNPredictor(
-            in_features, num_classes
+        super().__init__(*args, **kwargs)
+
+        # Define os pesos corretamente
+        if num_classes != 91:
+            weights = None
+
+        # Verifica se há pesos para o backbone nos argumentos
+        weights_backbone = kwargs.get("weights_backbone", None)
```

Comment:
deixa como argumento no init mesmo pow

```suggestion
        def __init__(
        self,
        num_classes: int = 9,
        weights: Optional[str] = 'DEFAULT',
        optimizer_name: str = 'adam',
        lr: float = 0.0001,
        weight_decay: float = 0.0001,
        weights_backbone: Optional = None,
        *args,
        **kwargs,
    ):
```

Reply:
Se eu deixar com os pesos DEFAULT ele define automaticamente o número de classes que serão utilizadas igual a 91 dentro da RCNN.